scrapper/utils/logger.py

```
@@ -0,0 +1,112 @@
+import logging
+import json
+from typing import Any, Optional
+from datetime import datetime
+
+
+class JsonFormatter(logging.Formatter):
+    """Custom JSON formatter for structured logging."""
+    
+    def format(self, record: logging.LogRecord) -> str:
+        log_data = {
+            "timestamp": datetime.utcnow().isoformat(),
```

Ruff (0.15.12)

[warning] `datetime.datetime.utcnow()` used

(DTZ003)